src/SdssShape.cc

```diff
@@ -882,5 +882,53 @@ INSTANTIATE_PIXEL(int);
 INSTANTIATE_PIXEL(float);
 INSTANTIATE_PIXEL(double);
 
-}}} // end namespace lsst::meas::base
+SdssShapeTransform::SdssShapeTransform(
+    Control const & ctrl,
+    std::string const & name,
+    afw::table::SchemaMapper & mapper
+) :
+    BaseTransform{name},
+    _fluxTransform{name, mapper},
+    _centroidTransform{name, mapper}
+{
+    for (auto flag = flagDefs.begin() + 1; flag < flagDefs.end(); flag++) {
+        mapper.addMapping(mapper.getInputSchema().find<afw::table::Flag>(
+                          mapper.getInputSchema().join(name, flag->name)).key);
+    }
+
+    _outShapeKey = ShapeResultKey::addFields(mapper.editOutputSchema(), name, "Shape in celestial moments",
+                                             FULL_COVARIANCE, afw::table::CoordinateType::CELESTIAL);
+}
+
+void SdssShapeTransform::operator()(
+    afw::table::SourceCatalog const & inputCatalog,
+    afw::table::BaseCatalog & outputCatalog,
+    afw::image::Wcs const & wcs,
+    afw::image::Calib const & calib
+) const {
+    _fluxTransform(inputCatalog, outputCatalog, wcs, calib);
+    _centroidTransform(inputCatalog, outputCatalog, wcs, calib);
 
+    CentroidResultKey centroidKey(inputCatalog.getSchema()[_name]);
+    ShapeResultKey inShapeKey(inputCatalog.getSchema()[_name]);
+
+    afw::table::SourceCatalog::const_iterator inSrc = inputCatalog.begin();
+    afw::table::BaseCatalog::iterator outSrc = outputCatalog.begin();
+    for (; inSrc < inputCatalog.end() && outSrc < outputCatalog.end(); ++inSrc, ++outSrc) {
```

Comment:
Docs claim this throws `LengthError` if sizes don't match; this looks like you're just using the smaller of the two sizes in this case.  Maybe better to check for that and throw up front, then only check one iterator in the loop iteration.